### DISCUSSION TITLE
How does concourse know about resource types?

### DISCUSSION THREAD

**Discussion by kkarthik23** [concourse/concourse#6611] [Help & Support] [answered]
Hi,
So i installed concourse on linux  using postgres as backend db and i have a web and worker node on the same server

i can get to the web interface  and login and see pipelines and used fly to create a pipeline however ran into some problems

I used https://concourse-ci.org/hello-world-example.html as a test however when i run it i get the error 
base resource type not found registry image

i used the book lit example and this time same issue but with git

Much appreciated if someone can answer this and help me fix this.

**Comment by evanchaoli** [comment, marked as answer]
`registry-image` and `git` are built-in resource types. They are natively installed together with worker. The problem may indicate some installation problem of worker.

**Reply by kkarthik23** [reply]
thanks for the reply   fly -t  ws shows the worker is running also the concourse worker is running as a service and there are no errors there.
When I look at the old discussion forums there are a few people who have had similar problems but there does not seem to be any answers to their questions surely there must be someone out there running concourse successfully on Linux  most people seem to be running on docker using the quick start docker compose method without issues and that works for me too but running it on a VM seems to be throwing this.  I have the resources in the concourse worker 
working directory and even that didn't help hence my question on how concourse knows about the resources.

**Reply by evanchaoli** [reply]
When a worker starts, it will scan assert directory, default to `/usr/local/concourse`, for resource types, and store found resource types to the db table `worker_base_resource_types`. Then ATC knows which workers have needed base resource types in order to run a build.

**Reply by taylorsilva** [reply]
To help troubleshoot, you can see what resources a worker has by adding `-d` to the `fly workers` command:

```
$ fly -t ci workers -d
name               containers  platform  tags  team ...   resource types
ci-worker-0        61          linux     none  none ...   bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, mock, pool, registry-image, s3, semver, time, tracker
ci-worker-1        58          linux     none  none ...   bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, mock, pool, registry-image, s3, semver, time, tracker
ci-worker-2        52          linux     none  none ...   bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, mock, pool, registry-image, s3, semver, time, tracker
ci-worker-3        65          linux     none  none ...   bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, mock, pool, registry-image, s3, semver, time, tracker
ci-worker-4        41          linux     none  none ...   bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, mock, pool, registry-image, s3, semver, time, tracker
ci-worker-5        55          linux     none  none ...   bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, mock, pool, registry-image, s3, semver, time, tracker
ci-worker-6        48          linux     none  none ...   bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, mock, pool, registry-image, s3, semver, time, tracker
ci-worker-7        59          linux     none  none ...   bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, mock, pool, registry-image, s3, semver, time, tracker
windows-worker     1           windows   none  main ...   none

```

**Reply by kkarthik23** [reply]
Working now this error is gone answer from @evanchaoli  helped me on the right track but i have a different error with dns i think the container has a problem reaching out to the docker registry  need to look at that 
thank you very much
